src/storage/src/source/postgres/replication.rs

```diff
@@ -355,21 +360,7 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                                         data_output.give(data_cap, update).await;
                                     }
                                 }
-                                assert!(
-                                    new_upper <= commit_lsn,
-                                    "new_upper={} tx_lsn={}",
-                                    new_upper,
-                                    commit_lsn
-                                );
-                                container.push((data, commit_lsn, diff));
```

Comment:
we were also buffering the entire transaction pre-reclocking here, apparently

Reply:
Yes, oops!